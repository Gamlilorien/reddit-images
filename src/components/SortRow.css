.sort-col {
    padding-bottom: 4px;
}
.not-active {
    color: #00CCCC;
  }
.not-active:hover {
color: #99FFFF;
}