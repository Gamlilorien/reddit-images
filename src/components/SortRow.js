import React from 'react';
import {Row, Col} from 'reactstrap';
import './SortRow.css';

const SortRow = (props) => {
    const sortState = props.sort;
    //console.log(sortState);
    //const [sort, updateSort] = useState(props.sort);

    const sortEntries = (event) => {
      //console.log(event.target.getAttribute('sortvalue'))
      let newValue = event.target.getAttribute('sortvalue');
      props.onSortHandler(newValue);
    }

    return (
        <Row sortvalue={sortState}>
            <Col className="sort-col">
            <div className="float-end">
                <span >Sort by: </span>
                <span sortvalue="post" className={sortState === 'post'?"active":"not-active"} onClick={sortEntries}>Post |</span>
                <span sortvalue="ascending" className={sortState === 'ascending'?"active":"not-active"} onClick={sortEntries}> A-z |</span>
                <span sortvalue="descending" className={sortState === 'descending'?"active":"not-active"} onClick={sortEntries}> Z-a</span>
                </div>
            </Col>
      </Row>
    )
}

export default SortRow;