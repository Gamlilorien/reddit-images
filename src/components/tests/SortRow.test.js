import React from 'react';
import renderer from 'react-test-renderer';
import * as ReactDomClient from 'react-dom/client';
import SortRow from '../SortRow';

const div = document.createElement("div");
const root = ReactDomClient.createRoot(div);

//make sure that the element gets created regarless of props
it("SortRow element renders", () => {
    root.render(<SortRow />)
})

//Compare with known good snapshot given same 'post' input
it("SortRow matches test snapshot", () => {
    const tree = renderer.create(<SortRow sort="post" />)
    expect(tree).toMatchSnapshot();
})