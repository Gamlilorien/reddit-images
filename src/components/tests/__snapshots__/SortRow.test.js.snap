// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortRow matches test snapshot 1`] = `
<div
  className="row"
  value="post"
>
  <div
    className="sort-col col"
  >
    <div
      className="float-end"
    >
      <span>
        Sort by: 
      </span>
      <span
        className="active"
        onClick={[Function]}
        value="post"
      >
        Post |
      </span>
      <span
        className="not-active"
        onClick={[Function]}
        value="ascending"
      >
         A-z |
      </span>
      <span
        className="not-active"
        onClick={[Function]}
        value="descending"
      >
         Z-a
      </span>
    </div>
  </div>
</div>
`;
